src/moi_wrapper.jl

```diff
@@ -165,41 +132,18 @@ function MOI.supports(model::Optimizer, attr::MOI.AbstractModelAttribute)
     return MOI.supports(model.optimizer, attr)
 end
 
-function MOI.supports(
-    model::Optimizer,
-    attr::MOI.ObjectiveFunction{<: SUPPORTED_OBJECTIVES}
-)
+function MOI.supports(model::Optimizer, attr::MOI.ObjectiveFunction)
     return MOI.supports(model.optimizer, attr)
 end
 
 function MOI.supports_constraint(
-    model::Optimizer,
-    ::Type{F},
-    ::Type{S}
-) where {
-    F <: SUPPORTED_SCALAR_FUNCTIONS,
-    S <: SUPPORTED_SCALAR_SETS,
-}
-    MOI.supports_constraint(model.optimizer, F, S)
-end
-function MOI.supports_constraint(
-    model::Optimizer,
-    ::Type{F},
-    ::Type{S}
-) where {
-    F <: SUPPORTED_VECTOR_FUNCTIONS,
-    S <: SUPPORTED_VECTOR_SETS,
-}
+    model::Optimizer, ::Type{F}, ::Type{S},
+) where {F<:MOI.AbstractFunction,S<:MOI.AbstractSet}
```

Comment:
we might need to check both optimizer and the underlying diff_models

Reply:
ping @blegat ?

Reply:
I'd rather add special cases for the attributes that are used with the `DiffOpt.AbstractModel`, that's what I just did in the last commit